server/repository/src/lib.rs

```diff
@@ -44,21 +44,13 @@ pub fn run_db_migrations(connection: &StorageConnection) -> Result<(), String> {
 
 use std::fmt::Debug as DebugTrait;
 pub trait Delete: DebugTrait {
-    fn delete(&self, con: &StorageConnection) -> Result<(), RepositoryError>;
+    fn delete(&self, con: &StorageConnection) -> Result<Option<i64>, RepositoryError>;
     // Test only
     fn assert_deleted(&self, con: &StorageConnection);
 }
 
 pub trait Upsert: DebugTrait {
-    // TODO:(Long term) DELETE THIS TRAIT METHOD AND REMOVE TRIGGERS?
-    fn upsert_sync(&self, con: &StorageConnection) -> Result<(), RepositoryError>;
-
-    fn upsert(&self, con: &StorageConnection) -> Result<Option<i64>, RepositoryError> {
-        self.upsert_sync(con)?;
-        // When not using triggers to create changelog records, this is where you may want to implement changelog logic
-        // This function should return the id of the changelog record created...
-        Ok(None)
-    }
+    fn upsert(&self, con: &StorageConnection) -> Result<Option<i64>, RepositoryError>;
```

Comment:
I think a small comment about Option<i64> wouel be great in the trait description